### DISCUSSION TITLE
Prisma `create` expects `id` field even it is an `autoincrement()` value

### DISCUSSION THREAD

**Discussion by joaocasarin** [prisma/prisma#14350] [Q&A] [answered]
So, I'm back trying to learn Prisma and I can't make it works with the testing guide.

I just created this setup:

Prisma models:
```prisma
model User {
    id         Int    @id @default(autoincrement())
    name       String
    email      String @unique
    password   String
    posts      Post[] @relation("CreatedBy")
    likedPosts Post[] @relation("LikedBy")
}

model Post {
    id        Int      @id @default(autoincrement())
    text      String
    createdAt DateTime @default(now())
    author    User     @relation("CreatedBy", fields: [authorId], references: [id])
    authorId  Int
    likedBy   User[]   @relation("LikedBy")
}
```

interfaces:
```ts
export interface User {
    name: string;
    email: string;
    password: string;
}

export interface UserResponse extends User {
    id: number;
}
```

The `User` service:
```ts
import prisma from '@database/client';
import { User, UserResponse } from '@interfaces/User';

export const createUser = async ({ name, email, password }: User): Promise<UserResponse | null> => {
    const isUserInDatabase = await prisma.user.findFirst({ where: { email } });

    if (isUserInDatabase) {
        return null;
    }

    const createdUser = await prisma.user.create({
        data: {
            name,
            email,
            password
        }
    });

    return createdUser;
};
```

With the following test:
```ts
import prismaMock from '@tests/singleton';
import { createUser } from '@services/User';
import { User } from '@interfaces/User';

test('should create new user ', async () => {
    const user: User = {
        name: 'Joao',
        email: 'joao@casarin.com',
        password: '12345678'
    };

    prismaMock.user.create.mockResolvedValue(user);

    await expect(createUser(user)).resolves.toEqual({
        id: 1,
        name: 'Joao',
        email: 'joao@casarin.com',
        password: '12345678'
    });
});
```

Why am I getting following error? It works when I send `{id: 1, ...user}` instead of just `user`. Why is the `create` method expecting the `id` field even though it is auto generated in the database?
```bash
Argument of type 'User' is not assignable to parameter of type 'User | Prisma__UserClient<User>'.
  Property 'id' is missing in type 'import("c:/Users/joaov/Desktop/express-k8s/src/interfaces/User").User' but required in type 'import("C:/Users/joaov/Desktop/express-k8s/node_modules/.prisma/client/index").User'.
```

A screenshot to better show you the problem:
![image](https://user-images.githubusercontent.com/48847394/179447059-5b00734c-99be-495a-a6c5-8b07a2932968.png)

PS.: The path mapping is working fine, so don't worry about using `@tests/singleton`, `@database/client` or whatever.

Thanks

**Comment by joaocasarin** [comment, marked as answer]
Nevermind. Just realized the `id` is being expected because I'm mocking the `response` of the `create` method, not the method arguments.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
